### PR TITLE
feat!: add table format to 'coder license ls'

### LINKED ISSUE #7697: feat: add table format for `coder license list`
Display the output using the `table` format as we do in other CLI lists.

discussed on: https://github.com/coder/coder/issues/4313#issuecomment-1552371762 and https://github.com/coder/coder/pull/7687#pullrequestreview-1445191712

### PR DESCRIPTION
**Pull request by Emyrk**
# What this does

Closes https://github.com/coder/coder/issues/7697

```bash
UUID                                  UPLOADED AT           FEATURES      EXPIRES AT                 
c9d1cbfa-6a8c-49f7-8999-82fc64934aef  2023-02-03T18:24:27Z  all features  2024-02-03T13:21:02-05:00 
```

If anyone has an idea on a better way to format the claims, lmk.

We might want to just move the License struct to the `codersdk` so we can json unmarshal the claims easier.

# Behavior changes

- We changed the `license_expires` claim in the json payload from a unix timestamp to a RFC3339 string in the `-o json` output. I kept this, but moved it to `license_expires_human`. I did this to maintain the original rational, but also preserve the original payload. The original reasoning was probably because the table view did not exist.
- Table view is now the default. `coder licenses ls -o json` for the json output.

### CODE REVIEW

File: enterprise/cli/licenses.go [hunk 1]
```diff
@@ -136,6 +137,71 @@ func validJWT(s string) error {
...
+				out := make([]tableLicense, 0, len(list))
+				for _, lic := range list {
+					var formattedFeatures string
+					features, err := lic.FeaturesClaims()
+					if err != nil {
+						formattedFeatures = xerrors.Errorf("invalid license: %w", err).Error()
+					} else {
+						var strs []string
+						if lic.AllFeaturesClaim() {
+							strs = append(strs, "all")
```
Comment (mafredri) [inline enterprise/cli/licenses.go:168]: Do we need to add all zero features if we add all?

```
UUID                                  UPLOADED AT           FEATURES                                                                           EXPIRES AT
1d6fab30-b9fe-40f1-8510-f054bbe252ab  2022-11-05T16:45:44Z  audit_log=1, user_limit=5                                                          2023-08-26T22:19:20Z
6a6fa845-b022-4980-953f-982617bdf74e  2023-01-19T22:01:43Z  all, workspace_quota=0, audit_log=0, browser_only=0, rbac=0, scim=0, user_limit=0  2023-10-10T21:01:34Z
```

In this case, I think simply omitting those would make sense: `all, workspace_quota=0, audit_log=0, browser_only=0, rbac=0, scim=0, user_limit=0` -> `all`.

File: codersdk/licenses.go [hunk 2]
```diff
@@ -23,11 +27,49 @@ type License struct {
...
+		if err != nil {
+			return time.Time{}, xerrors.Errorf("license_expires claim is not a valid unix timestamp: %w", err)
+		}
+		return time.Unix(i64, 0), nil
+	}
+
+	return time.Time{}, xerrors.Errorf("license_expires claim has unexpected type %T", expClaim)
+}
+
+func (l *License) Trail() bool {
```
Comment (mafredri) [inline codersdk/licenses.go:55]: `s/Trail/Trial/g`

File: enterprise/cli/licenses.go [hunk 3]
```diff
@@ -136,6 +137,71 @@ func validJWT(s string) error {
...
+			}),
+		cliui.ChangeFormatterData(cliui.JSONFormat(), func(data any) (any, error) {
+			list, ok := data.([]codersdk.License)
+			if !ok {
+				return nil, xerrors.Errorf("invalid data type %T", data)
+			}
+			for i := range list {
+				humanExp, err := list[i].ExpiresAt()
+				if err == nil {
+					list[i].Claims[codersdk.LicenseExpiryClaim+"_human"] = humanExp.Format(time.RFC3339)
```
Comment (mafredri) [inline enterprise/cli/licenses.go:197]: I'd say both this and the change in default output would warrant a breaking change.

**Review by mafredri — APPROVED** [review]
Some minor nits, but otherwise looks good!

File: codersdk/licenses.go (hunk 2 above)
Comment (Emyrk) [reply codersdk/licenses.go:55]: :facepalm:

File: enterprise/cli/licenses.go (hunk 3 above)
Comment (Emyrk) [reply enterprise/cli/licenses.go:197]: Fair :+1:

File: enterprise/cli/licenses.go (hunk 1 above)
Comment (Emyrk) [reply enterprise/cli/licenses.go:168]: I was honestly curious why we even include the claim if it's 0.

You are right though, I'll omit 0s.

**Pull request merged**